src/protocols/coda_pow.ml

```diff
@@ -935,42 +935,44 @@ module type Transaction_snark_scan_state_intf = sig
 
   val empty : unit -> t
 
-  val capacity : t -> int
+  (*val capacity : t -> int*)
 
   val fill_work_and_enqueue_transactions :
        t
     -> Transaction_with_witness.t list
     -> transaction_snark_work list
-    -> (ledger_proof * transaction list) option Or_error.t
+    -> ((ledger_proof * transaction list) option * t) Or_error.t
 
   val latest_ledger_proof :
     t -> (Ledger_proof_with_sok_message.t * transaction list) option
 
   val free_space : t -> int
 
-  val next_k_jobs : t -> k:int -> Available_job.t list Or_error.t
+  val next_k_jobs : t -> k:int -> Available_job.t list list Or_error.t
 
-  val next_jobs : t -> Available_job.t list Or_error.t
+  val next_jobs : t -> Available_job.t list list
```

Comment:
what is the second nested list for?

Reply:
Because the jobs could be from multiple trees and proofs bundles should be created for each tree so that the pairing is same always. For example if `[p1;p2;p3]` and `[p1'; p2']` are the jobs for two different trees that need to be completed for the current diff, then they should be bundled as `[p1;p2]`, `[p3]`, `[p1';p2']` and not as `[p1;p2]`, `[p3;p1']`, `[p2']`. This ensures that the proof bundles always remain the same and doesn't change with the throughput like it did in the previous implementation. The comment for this in `Transaction_snark_scan_state` where the function is defined

Reply:
awesome, can you move the comment to the mli with `(**` comments